contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2989,7 +2993,13 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 				DropStmt   *drop_stmt = (DropStmt *) parsetree;
 
 				if (drop_stmt->removeType != OBJECT_SCHEMA)
+				{
+					if (sql_dialect == SQL_DIALECT_TSQL)
+						bbf_ExecDropStmt(drop_stmt);
```

Comment:
Here we cleaning up the extended property catalog without making sure that statement execution is successful or not. Please consider the scenario that what if drop statement fail due to some error like permission error or anything else - ideally catalog shouldn't be cleaned up in that case but it entry will be removed.

Reply:
Firstly, dropping catalog and dropping statement are in the same transaction. If dropping statement fails, dropping catalog will be rolled back.
Secondly, if dropping statement fails due to permission error, it's very likely that dropping catalog will fail due to permission error, too.

Reply:
Got it. Wanted to confirm 1st point.

Reply:
1st point is based on the MVCC realization of PG.
Simple verification is to add elog(ERROR, ...) after bbf_ExecDropStmt, and you will see the same result as I said.